jhdf/src/main/java/io/jhdf/AbstractNode.java

```diff
@@ -116,7 +116,7 @@ public String getName() {
 
 	@Override
 	public String getPath() {
-		return parent.getPath() + name;
+		return parent == null ? null : parent.getPath() + name;
```

Comment:
If you have a reference and get an address from it, you can do: 
`ObjectHeader linkHeader = ObjectHeader.readObjectHeader(hdfFc, address);
DatasetLoader.createDataset(hdfFc, linkHeader, null, null);
`
Then you have a dataset that you can read, but you don't know its parent or name.
Theoretically, a Reference can point to a header that is not listed in any group.
Do you have any other suggestions what we could do here? With this fix, the toString method will no longer result in an NPE.

Reply:
Ok I think it would be better to make a special `NoParent` class implementing `Group` for this use case. And just throw in almost all the methods except `getPath` where you could return something like `Unknown` or `No parent`.

Reply:
Sounds like a good idea